internal/catalog/exports.go

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package catalog
+
+import (
+	"github.com/hashicorp/consul/internal/catalog/internal/controllers"
+	"github.com/hashicorp/consul/internal/catalog/internal/types"
+	"github.com/hashicorp/consul/internal/controller"
+	"github.com/hashicorp/consul/internal/resource"
+)
+
+var (
+	// Resource Types for the v1alpha1 version. These are mainly applicable for
+	// code to implement migration/conversion logic. In most cases the current
+	// version types should be used instead to represent the latest type
+
+	WorkloadV1Alpha1Type         = types.WorkloadV1Alpha1Type
+	ServiceV1Alpha1Type          = types.ServiceV1Alpha1Type
+	ServiceEndpointsV1Alpha1Type = types.ServiceEndpointsV1Alpha1Type
+	VirtualIPsV1Alpha1Type       = types.VirtualIPsV1Alpha1Type
+	NodeV1Alpha1Type             = types.NodeV1Alpha1Type
+	HealthStatusV1Alpha1Type     = types.HealthStatusV1Alpha1Type
+	HealthChecksV1Alpha1Type     = types.HealthChecksV1Alpha1Type
+	DNSPolicyV1Alpha1Type        = types.DNSPolicyV1Alpha1Type
+
+	// Resource Types for the current version. These types should be the
+	// ones generally used by controllers and other things in the system
```

Comment:
I think that's fine for controllers owned by the same team as the resource type (i.e. the core catalog/mesh controllers) but I don't think it's a pattern we'd want to encourage for _other_ teams' controllers that consume your resources, because it creates unnecessary organizational coupling.

If the external team "pins" to `v1` of your resource type, the version translation will ensure it keeps on working when you change the "current" version to `v2` - giving the external team time to upgrade. If they were using the "current" version constant instead, their controller would immediately break, so you'd either need to do the upgrade for them or work with them to get it fixed.